### ISSUE TITLE
Create a new coder template for kubevirt

### ISSUE THREAD

**Issue by hh** [sharingio/pair#187]
- Talos + Equinix will have a cost of a machine per user.
- vcluster is quite insecure
- KubeVirt might be a decent middle ground allowing for better allocation of real hardware for multiple developers

**Comment by hh** [comment]
Initial stab:
https://github.com/sharingio/coder/tree/main/examples/templates/kubevirt

**Comment by hh** [comment]
Next step is CNI: https://github.com/sharingio/coder/blob/main/examples/templates/kubevirt/Readme.md#cni-not-yet-working

**Comment by hh** [comment]
I wonder, if for speed, talos might be a better image to use.... and avoid using the kubeadm based approach.

See https://twitter.com/usrbinkat/status/1516590124963835905

**Comment by hh** [comment]
Caleb can we do this without clusterapi?

**Comment by BobyMCbobs** [comment]
> Caleb can we do this without clusterapi?

Yes it is possible, but the amount of things required to get it going is a lot more.
There'll be a lot more config, specifically around things like userdata init and such.
I can take a look, but Cluster-API is the clearest path with least config as far as i'm aware.
